Description: permit Mod and CreDate pdfmarks in PDF 2.0 in pdfwrite
 Bug 699807
 "-dCompatibilityLevel=2.0` prevents all metadata from being set"
 .
 PDF 2.0 has deprecated The Document Info dictionary,
 so we prevented pdfmarks from modifying the data in it,
 as a prelude to removing it potentially
 when later versions of the PDF spec are approved.
 .
 However, as the report says,
 the Creation Date and Modification Date are still required
 for documents with PieceInfo (only),
 so for now we will permit those two values (only) to be set.
Origin: upstream, http://git.ghostscript.com/?p=ghostpdl.git;h=863b370
Author: Ken Sharp <ken.sharp@artifex.com>
Forwarded: yes
Bug: http://bugs.ghostscript.com/show_bug.cgi?id=699807
Last-Update: 2018-10-18
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/devices/vector/gdevpdfm.c
+++ b/devices/vector/gdevpdfm.c
@@ -1970,9 +1970,6 @@
     int code = 0, i;
     gs_memory_t *mem = pdev->pdf_memory;
 
-    if (pdev->CompatibilityLevel >= 2.0)
-        return 0;
-
     if (count & 1)
         return_error(gs_error_rangecheck);
     for (i = 0; code >= 0 && i < count; i += 2) {
@@ -1984,6 +1981,11 @@
 
         vsize = 0x0badf00d; /* Quiet compiler. */
 
+        if (pdev->CompatibilityLevel >= 2.0) {
+            if (!pdf_key_eq(pairs + i, "/ModDate") && !pdf_key_eq(pairs + i, "/CreationDate"))
+                continue;
+        }
+
         if (pdev->PDFA !=0) {
             const gs_param_string *p = pairs + i + 1;
             if (p->size > 9 && memcmp(p->data, "(\\376\\377", 9) == 0) {
